lib/paths.coffee

```diff
@@ -0,0 +1,144 @@
+# Port of jupyter_core.paths
+
+fs = require('fs')
+path = require('path')
+
+# Access `sys.prefix` from Python, to handle particular conda and virtualenv setups
+# TODO: Think of something more sensible here, possibly doing this asynchronously elsewhere
+# TODO: Provide a timeout, handle error
+{execSync} = require('child_process')
+response = execSync 'python -c "import sys; print(sys.prefix)"'
+sysPrefix = response.toString().replace /^\s+|\s+$/g, ""
+
+# Compute default system configurations
+if process.platform == 'win32'
+  programData = process.env['PROGRAMDATA']
+  if programData
+    SYSTEM_JUPYTER_PATH = [path.join(programData, 'jupyter')]
+  else
+    SYSTEM_JUPYTER_PATH = [path.join(sysPrefix, 'share', 'jupyter')]
+else
+  SYSTEM_JUPYTER_PATH = [
+    "/usr/local/share/jupyter",
+    "/usr/share/jupyter"
+  ]
+
+ENV_JUPYTER_PATH = [path.join(sysPrefix, 'share', 'jupyter')]
+
+# Returns the home specified by environment variable or
+# node's built in path.resolve('~')
+userHome = ->
+  homeDir = process.env['HOME'] or process.env['USERPROFILE'] or path.resolve('~')
+  homeDir = fs.realpathSync(homeDir)
+  return homeDir
+
+# Get the Jupyter config directory for this platform and user.
+# Reutrns env[JUPYTER_CONFIG_DIR] if defined, else ~/.jupyter
+jupyterConfigDir = ->
+  homeDir = userHome()
+  if process.env['JUPYTER_CONFIG_DIR']
+    return process.env['JUPYTER_CONFIG_DIR']
+  return path.join(homeDir, '.jupyter')
+
+# Get the config directory for Jupyter data files.
+#
+# These are non-transient, non-configuration files.
+#
+# Returns process.env[JUPYTER_DATA_DIR] if defined,
+# else a platform-appropriate path.
+jupyterDataDir = ->
+  if process.env['JUPYTER_DATA_DIR']
+    return process.env['JUPYTER_DATA_DIR']
+
+  # Windows first for easier testing
+  if process.platform == 'win32'
+    appData = process.env['APPDATA']
+    if appData
+      return path.join appData, 'jupyter'
+    else
+      return path.join jupyterConfigDir(), 'data'
+
+  home = userHome()
```

Comment:
Let's make this an `if... else if... else` instead of relying on the `return`s to be more explicit about the control flow